sys/net/sock/sock_util.c

```diff
@@ -116,30 +117,35 @@ static char* _find_pathstart(const char *url)
 
 int sock_urlsplit(const char *url, char *hostport, char *urlpath)
 {
+    assert(url);
     char *hoststart = _find_hoststart(url);
     if (!hoststart) {
         return -EINVAL;
     }
 
     char *pathstart = _find_pathstart(hoststart);
 
-    size_t hostlen = pathstart - hoststart;
-    /* hostlen must be smaller SOCK_HOSTPORT_MAXLEN to have space for the null
-     * terminator */
-    if (hostlen > SOCK_HOSTPORT_MAXLEN - 1) {
-        return -EOVERFLOW;
+    if (hostport) {
+        size_t hostlen = pathstart - hoststart;
+        /* hostlen must be smaller SOCK_HOSTPORT_MAXLEN to have space for the null
+        * terminator */
+        if (hostlen > SOCK_HOSTPORT_MAXLEN - 1) {
+            return -EOVERFLOW;
+        }
+        memcpy(hostport, hoststart, hostlen);
+        *(hostport + hostlen) = '\0';
     }
-    memcpy(hostport, hoststart, hostlen);
-    *(hostport + hostlen) = '\0';
 
-    size_t pathlen = strlen(pathstart);
-    if (pathlen) {
-        if (pathlen > SOCK_URLPATH_MAXLEN - 1) {
-            return -EOVERFLOW;
+    if (urlpath) {
+        size_t pathlen = strlen(pathstart);
+        if (pathlen) {
+            if (pathlen > SOCK_URLPATH_MAXLEN - 1) {
+                return -EOVERFLOW;
+            }
+            memcpy(urlpath, pathstart, pathlen);
         }
-        memcpy(urlpath, pathstart, pathlen);
+        *(urlpath + pathlen) = '\0';
```

Comment:
```suggestion
        urlpath[pathlen] = '\0';
```
same as above, since you are you are touching the code, why not fix this too.